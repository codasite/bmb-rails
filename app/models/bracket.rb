class Bracket < ApplicationRecord
  validates :title, presence: true
  validates :slug, presence: true, uniqueness: true
  validates :num_teams, presence: true, numericality: { greater_than: 0 }

  before_validation :generate_slug, if: :title_changed?

  serialize :round_names, coder: JSON

  scope :by_year, ->(year) { where(year: year) if year.present? }
  scope :by_month, ->(month) { where(month: month) if month.present? }
  scope :voting, -> { where(is_voting: true) }
  scope :tournament, -> { where(is_voting: false) }

  has_many :bracket_matches, dependent: :destroy
  has_many :picks, dependent: :destroy
  has_many :plays, dependent: :destroy

  def to_param
    slug
  end

  def voting?
    is_voting
  end

  def tournament?
    !is_voting
  end

  def has_fee?
    fee.present? && fee > 0
  end

  def current_round
    live_round_index
  end

  def completed?
    # Logic to determine if bracket is completed
    # This would depend on your specific business rules
    false
  end

  private

  def generate_slug
    self.slug = title.parameterize if title.present?
  end
end
