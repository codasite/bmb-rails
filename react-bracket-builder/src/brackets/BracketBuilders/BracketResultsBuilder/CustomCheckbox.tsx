import checkIcon from '../../shared/assets/check.svg'

export interface CustomCheckboxProps {
  id: string
  checked: boolean
  onChange: (e: any) => void
  height?: number
  width?: number
}

export const CustomCheckbox = (props: any) => {
  const { id, checked, onChange, height = 32, width = 32 } = props

  const baseStyles = ['tw-appearance-none', 'tw-rounded-8', 'tw-cursor-pointer']

  const uncheckedStyles = ['tw-border', 'tw-border-solid', 'tw-border-white']

  const checkedStyles = ['tw-bg-white', 'tw-bg-no-repeat', 'tw-bg-center']

  const styles = baseStyles
    .concat(checked ? checkedStyles : uncheckedStyles)
    .join(' ')

  return (
    <input
      type="checkbox"
      id={id}
      className={styles}
      checked={checked}
      onChange={onChange}
      style={{
        backgroundImage: checked ? `url(${checkIcon})` : 'none',
        height,
        width,
      }}
    />
  )
}
