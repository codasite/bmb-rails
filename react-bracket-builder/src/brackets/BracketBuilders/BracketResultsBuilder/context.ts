import { createContext } from 'react'

export interface BracketResultsBuilderContextState {
  notifyParticipants?: boolean
  toggleNotifyParticipants?: () => void
}

export const BracketResultsBuilderContext =
  createContext<BracketResultsBuilderContextState>({})
