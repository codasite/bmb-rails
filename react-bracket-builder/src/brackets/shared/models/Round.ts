import { Nullable } from '../../../utils/types'
import { MatchNode } from './operations/MatchNode'
import { MatchRepr } from '../api/types/bracket'
import { Team } from './Team'

export class Round {
  index: number
  depth: number
  matches: Array<Nullable<MatchNode>>
  constructor(
    index: number,
    depth: number,
    matches: Array<Nullable<MatchNode>> = []
  ) {
    this.index = index
    this.depth = depth
    this.matches = matches
  }
  allPicked(): boolean {
    return this.matches.every((match) => {
      if (match === null) {
        return true
      }
      return match.team1Wins || match.team2Wins
    })
  }
  getTeam(teamId: number): Nullable<Team> {
    for (const [index, match] of this.matches.entries()) {
      if (match === null) {
        continue
      }
      const team1 = match.getTeam1()
      const team2 = match.getTeam2()

      if (team1?.id === teamId) {
        team1.side = index < this.matches.length / 2 ? 'left' : 'right'
        return team1
      }

      if (team2?.id === teamId) {
        team2.side = index < this.matches.length / 2 ? 'left' : 'right'
        return team2
      }
    }

    return null
  }
  serialize(): Nullable<MatchRepr>[] {
    return this.matches.map((match, i) => {
      if (match === null) {
        return null
      }
      return match.serialize()
    })
  }
}
