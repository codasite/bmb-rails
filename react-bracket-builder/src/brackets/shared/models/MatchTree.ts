import { Nullable } from '../../../utils/types'
import { BracketRes, MatchPick, MatchReq, MatchTreeRepr } from '../api'
import { WildcardPlacement } from './WildcardPlacement'
import { Team } from './Team'
import { MatchNode } from './operations/MatchNode'
import { Round } from './Round'
import { matchReprFromNumTeams } from './operations/MatchReprFromNumTeams'
import { getNumRounds } from './operations/GetNumRounds'
import { matchReprFromRes } from './operations/MatchReprFromRes'
import { linkNodes } from './operations/LinkNodes'

export class MatchTree {
  rounds: Round[]
  private numTeams: number
  private wildcardPlacement?: WildcardPlacement
  isVoting: boolean
  liveRoundIndex: number

  constructor(
    rounds: Round[] = [],
    wildcardPlacement?: WildcardPlacement,
    isVoting: boolean = false,
    liveRoundIndex: number = 0
  ) {
    linkNodes(rounds)
    this.rounds = rounds
    this.wildcardPlacement = wildcardPlacement
    this.isVoting = isVoting
    this.liveRoundIndex = liveRoundIndex
  }

  getRootMatch(): Nullable<MatchNode> {
    const lastRound = this.rounds[this.rounds.length - 1]
    if (!lastRound) {
      return null
    }
    return lastRound.matches[0]
  }

  clone(): MatchTree {
    return MatchTree.deserialize(this.serialize())
  }

  equals(other: MatchTree): boolean {
    return (
      JSON.stringify(this.serialize()) === JSON.stringify(other.serialize())
    )
  }

  /**
   * Returns the total number of POSSIBLE teams in the tournament.
   * This should reflect the number of teams that would be used to create the same structure with MatchTree.fromNumTeams()
   */
  getNumTeams(): number {
    if (this.numTeams) {
      return this.numTeams
    }
    // count all non null matches in the tree
    const numMatches = this.rounds.reduce((acc, round) => {
      return (
        acc +
        round.matches.reduce((acc, match) => {
          if (match) {
            return acc + 1
          }
          return acc
        }, 0)
      )
    }, 0)

    const numTeams = numMatches + 1
    this.numTeams = numTeams

    return numTeams
  }

  getWildcardPlacement(): WildcardPlacement | undefined {
    return this.wildcardPlacement
  }

  serialize(): MatchTreeRepr {
    const tree = this
    const rounds = tree.rounds.map((round) => {
      return round.serialize()
    })
    return {
      rounds: rounds,
      wildcardPlacement: tree.wildcardPlacement,
      isVoting: tree.isVoting,
      liveRoundIndex: tree.liveRoundIndex,
    }
  }

  advanceTeam = (
    roundIndex: number,
    matchIndex: number,
    left: boolean,
    requireComplete: boolean = false
  ) => {
    if (requireComplete && roundIndex > 0) {
      const prevRound = this.rounds[roundIndex - 1]
      if (prevRound && !prevRound.allPicked()) {
        return
      }
    }
    const round = this.rounds[roundIndex]
    const match = round.matches[matchIndex]
    if (!match) {
      return
    }
    match.team1Wins = false
    match.team2Wins = false
    if (left) {
      match.team1Wins = true
    } else {
      match.team2Wins = true
    }
  }

  anyPicked = (): boolean => {
    return this.rounds.some((round) => {
      return round.matches.some((match) => {
        if (!match) {
          return false
        }
        return match.team1Wins || match.team2Wins
      })
    })
  }

  allPicked = (): boolean => {
    if (this.isVoting) {
      return this.rounds[this.liveRoundIndex].allPicked()
    }
    return this.rounds.every((round) => {
      return round.allPicked()
    })
  }

  allTeamsAdded = (): boolean => {
    return this.everyMatch((match) => {
      let hasNeededTeams = true
      if (!match.left && !match.getTeam1()?.name) {
        hasNeededTeams = false
      }
      if (hasNeededTeams && !match.right && !match.getTeam2()?.name) {
        hasNeededTeams = false
      }
      return hasNeededTeams
    })
  }

  toMatchReq = (): MatchReq[] => {
    const matches: MatchReq[] = []
    this.forEachMatch((match, matchIndex, roundIndex) => {
      const matchReq: MatchReq = {
        roundIndex: match.roundIndex,
        matchIndex: match.matchIndex,
      }
      if (!match.left) {
        matchReq.team1 = match.getTeam1()?.toTeamReq()
      }
      if (!match.right) {
        matchReq.team2 = match.getTeam2()?.toTeamReq()
      }
      if (matchReq.team1 || matchReq.team2) {
        matches.push(matchReq)
      }
    })
    return matches
  }

  toMatchPicks = (): MatchPick[] => {
    const picks: MatchPick[] = []
    this.rounds.forEach((round, roundIndex) => {
      round.matches.forEach((match, matchIndex) => {
        if (!match) {
          return
        }
        const { team1Wins, team2Wins } = match
        const team1 = match.getTeam1()
        const team2 = match.getTeam2()
        const team1Winner = team1Wins && team1
        const team2Winner = team2Wins && team2

        if (team1Winner && team1.id) {
          picks.push({
            roundIndex,
            matchIndex,
            winningTeamId: team1.id,
          })
        } else if (team2Winner && team2.id) {
          picks.push({
            roundIndex,
            matchIndex,
            winningTeamId: team2.id,
          })
        }
      })
    })
    return picks
  }

  forEachMatch = (
    callback: (match: MatchNode, matchIndex: number, roundIndex: number) => void
  ) => {
    this.rounds.forEach((round, roundIndex) => {
      round.matches.forEach((match, matchIndex) => {
        if (!match) {
          return
        }
        callback(match, matchIndex, roundIndex)
      })
    })
  }

  everyMatch = (
    callback: (
      match: MatchNode,
      matchIndex: number,
      roundIndex: number
    ) => boolean
  ) => {
    return this.rounds.every((round, roundIndex) => {
      return round.matches.every((match, matchIndex) => {
        if (!match) {
          return true
        }
        return callback(match, matchIndex, roundIndex)
      })
    })
  }

  findMatch = (
    callback: (match: Nullable<MatchNode>) => boolean
  ): Nullable<MatchNode> => {
    let foundMatch: Nullable<MatchNode> = null
    this.rounds.some((round) => {
      return round.matches.some((match) => {
        if (callback(match)) {
          foundMatch = match
          return true
        }
        return false
      })
    })
    return foundMatch
  }

  static fromNumTeams(
    numTeams: number,
    wildcardPlacement: WildcardPlacement = WildcardPlacement.Top
  ): MatchTree {
    const matches = matchReprFromNumTeams(numTeams, wildcardPlacement)
    return MatchTree.deserialize({
      rounds: matches,
      wildcardPlacement,
    })
  }

  static fromMatchRes(
    bracket: Pick<
      BracketRes,
      'matches' | 'numTeams' | 'isVoting' | 'liveRoundIndex'
    >,
    wildcardPlacement?: WildcardPlacement
  ): MatchTree | null {
    const numRounds = getNumRounds(bracket.numTeams)

    const nestedMatches = matchReprFromRes(numRounds, bracket.matches)
    return MatchTree.deserialize({
      rounds: nestedMatches,
      wildcardPlacement,
      isVoting: bracket.isVoting,
      liveRoundIndex: bracket.liveRoundIndex,
    })
  }

  static fromPicks(
    bracket: Pick<
      BracketRes,
      'matches' | 'numTeams' | 'isVoting' | 'liveRoundIndex'
    >,
    picks: MatchPick[],
    wildcardPlacement?: WildcardPlacement
  ): MatchTree | null {
    const matchTree = MatchTree.fromMatchRes(bracket, wildcardPlacement)
    if (!matchTree) {
      return null
    }
    for (const pick of picks) {
      const { roundIndex, matchIndex, winningTeamId } = pick
      const match = matchTree.rounds[roundIndex].matches[matchIndex]
      if (!match) {
        throw new Error('Match not found')
      }
      match._pick = pick

      const team1 = match.getTeam1()
      const team2 = match.getTeam2()
      if (team1?.id === winningTeamId) {
        match.team1Wins = true
      } else if (team2?.id === winningTeamId) {
        match.team2Wins = true
      } else {
        throw new Error('Match does not have winningTeam from pick')
      }
    }
    return matchTree
  }

  static deserialize(matchTreeRepr: MatchTreeRepr): MatchTree {
    const { rounds: matchRes, wildcardPlacement } = matchTreeRepr
    const rounds = matchRes.map((round, roundIndex) => {
      const depth = matchRes.length - roundIndex - 1
      const newRound = new Round(roundIndex, depth)
      const matches = round.map((match, matchIndex) => {
        if (match === null) {
          return null
        }
        const {
          id,
          team1: team1Repr,
          team2: team2Repr,
          team1Wins,
          team2Wins,
          pick,
        } = match
        const team1 = team1Repr ? new Team(team1Repr.name, team1Repr.id) : null
        const team2 = team2Repr ? new Team(team2Repr.name, team2Repr.id) : null
        // const newMatch = new MatchNode(roundIndex, matchIndex, depth, match.id, team1, team2, team1Wins, team2Wins)
        const newMatch = new MatchNode({
          id,
          matchIndex,
          roundIndex,
          team1,
          team2,
          team1Wins,
          team2Wins,
          left: null,
          right: null,
          parent: null,
          depth,
          pick,
        })
        return newMatch
      })
      newRound.matches = matches
      return newRound
    })
    const tree = new MatchTree(
      rounds,
      wildcardPlacement,
      matchTreeRepr.isVoting,
      matchTreeRepr.liveRoundIndex
    )
    return tree
  }
}
