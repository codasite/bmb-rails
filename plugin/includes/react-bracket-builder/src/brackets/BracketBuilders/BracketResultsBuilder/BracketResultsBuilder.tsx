import React, { useEffect, useState } from 'react'
import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMetaContext, DarkModeContext } from '../../shared/context'
import darkBracketBg from '../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../shared/assets/bracket-bg-light.png'
import { ResultsBracket } from '../../shared/components/Bracket'
import { ActionButton } from '../../shared/components/ActionButtons'
import checkIcon from '../../shared/assets/check.svg'
import { bracketApi } from '../../shared/api/bracketApi'
import {
  WithBracketMeta,
  WithMatchTree,
  WithProvider,
} from '../../shared/components/HigherOrder'

const CustomCheckbox = (props: any) => {
  const { id, checked, onChange } = props

  const baseStyles = [
    'tw-appearance-none',
    'tw-h-32',
    'tw-w-32',
    'tw-rounded-8',
    'tw-cursor-pointer',
  ]

  const uncheckedStyles = ['tw-border', 'tw-border-solid', 'tw-border-white']

  const checkedStyles = ['tw-bg-white', 'tw-bg-no-repeat', 'tw-bg-center']

  const styles = baseStyles
    .concat(checked ? checkedStyles : uncheckedStyles)
    .join(' ')

  return (
    <input
      type="checkbox"
      id={id}
      className={styles}
      checked={checked}
      onChange={onChange}
      style={{ backgroundImage: checked ? `url(${checkIcon})` : 'none' }}
    />
  )
}

interface BracketResultsBuilderProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  bracket?: any
  myBracketsUrl?: string
}

const BracketResultsBuilder = (props: BracketResultsBuilderProps) => {
  const { matchTree, setMatchTree, bracket, myBracketsUrl } = props

  const [bracketTitle, setBracketTitle] = useState('')
  const [bracketMonth, setBracketMonth] = useState('')
  const [bracketYear, setBracketYear] = useState('')
  const [notifyParticipants, setNotifyParticipants] = useState(true)
  const [bracketId, setBracketId] = useState(0)
  console.log('bracket', bracket)

  useEffect(() => {
    if (bracket) {
      const numTeams = bracket.numTeams
      const matches = bracket.matches
      const results = bracket.results
      setBracketTitle(bracket.title)
      setBracketMonth(bracket.month)
      setBracketYear(bracket.year)
      setBracketId(bracket.id)
      let tree: MatchTree | null
      if (results && results.length > 0) {
        tree = MatchTree.fromPicks(numTeams, matches, results)
      } else {
        tree = MatchTree.fromMatchRes(numTeams, matches)
      }
      if (tree) {
        setMatchTree?.(tree)
      }
    }
  }, [])

  const darkMode = true
  const complete = matchTree && matchTree.allPicked()

  const handleUpdatePicks = () => {
    if (matchTree) {
      const picks = matchTree.toMatchPicks()
      if (!picks || picks.length === 0) return
      const complete = matchTree.allPicked()
      const data = {
        results: picks,
        updateNotifyParticipants: notifyParticipants,
      }
      bracketApi
        .updateBracket(bracketId, data)
        .then((res) => {
          console.log(res)
        })
        .catch((err) => {
          console.log(err)
        })
        .finally(() => {
          if (myBracketsUrl) window.location.href = myBracketsUrl || ''
        })
    }
  }

  return (
    <BracketMetaContext.Provider
      value={{ title: bracketTitle, month: bracketMonth, year: bracketYear }}
    >
      <DarkModeContext.Provider value={darkMode}>
        <div
          className={`wpbb-reset tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover${
            darkMode ? ' tw-dark' : ''
          }`}
          style={{
            backgroundImage: `url(${
              darkMode ? darkBracketBg : lightBracketBg
            })`,
          }}
        >
          {matchTree && (
            <div
              className={`tw-flex tw-flex-col tw-items-center tw-max-w-screen-lg tw-m-auto tw-pb-[57px]`}
            >
              <div className="tw-py-[116px]">
                <ResultsBracket
                  matchTree={matchTree}
                  setMatchTree={setMatchTree}
                />
              </div>
              <div
                className={`tw-flex tw-flex-col tw-gap-24${
                  !complete ? ' tw-max-w-[470px] tw-w-full' : ''
                }`}
              >
                <ActionButton variant="big-yellow" onClick={handleUpdatePicks}>
                  {complete ? 'Complete Bracket' : 'Update Picks'}
                </ActionButton>
                <div className="tw-flex tw-gap-20 tw-items-center tw-self-center">
                  <CustomCheckbox
                    id="notify-participants-check"
                    checked={notifyParticipants}
                    onChange={() => setNotifyParticipants(!notifyParticipants)}
                  />
                  <label
                    htmlFor="notify-participants-check"
                    className="tw-font-500 tw-text-24"
                  >
                    Notify Participants
                  </label>
                </div>
              </div>
            </div>
          )}
        </div>
      </DarkModeContext.Provider>
    </BracketMetaContext.Provider>
  )
}

const Wrapped = WithProvider(
  WithMatchTree(WithBracketMeta(BracketResultsBuilder))
)

export default Wrapped
