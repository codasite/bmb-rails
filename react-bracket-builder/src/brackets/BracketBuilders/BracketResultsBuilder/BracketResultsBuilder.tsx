import React, { useEffect, useState } from 'react'
import { logger } from '../../../utils/Logger'
import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMeta } from '../../shared/context/context'
import darkBracketBg from '../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../shared/assets/bracket-bg-light.png'
import { ResultsBracket } from '../../shared/components/Bracket'
import { ActionButton } from '../../shared/components/ActionButtons'
import { bracketApi } from '../../shared/api/bracketApi'
import {
  WithBracketMeta,
  WithDarkMode,
  WithMatchTree,
} from '../../shared/components/HigherOrder'
import {
  getBracketMeta,
  getBracketWidth,
} from '../../shared/components/Bracket/utils'
import { useWindowDimensions } from '../../../utils/hooks'
import { getNumRounds } from '../../shared/models/operations/GetNumRounds'
import { PaginatedResultsBuilder } from './PaginatedResultsBuilder/PaginatedResultsBuilder'
import { Checkbox } from './Checkbox'
import { BracketResultsBuilderContext } from './context'
import { getDashboardPath } from '../../shared'

interface BracketResultsBuilderProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  bracket?: any
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  darkMode?: boolean
  setDarkMode?: (darkMode: boolean) => void
}

const BracketResultsBuilder = (props: BracketResultsBuilderProps) => {
  const { matchTree, setMatchTree, bracket, setBracketMeta, darkMode } = props

  const [notifyParticipants, setNotifyParticipants] = useState(true)
  const [bracketId, setBracketId] = useState(0)

  const { width: windowWidth, height: windowHeight } = useWindowDimensions()
  const showPaginated =
    windowWidth - 100 < getBracketWidth(getNumRounds(bracket?.numTeams))

  const toggleNotifyParticipants = () => {
    setNotifyParticipants(!notifyParticipants)
  }

  useEffect(() => {
    if (bracket) {
      const numTeams = bracket.numTeams
      const matches = bracket.matches
      const results = bracket.results
      const meta = getBracketMeta(bracket)
      setBracketMeta?.(meta)
      setBracketId(bracket.id)
      let tree: MatchTree | null
      if (results && results.length > 0) {
        tree = MatchTree.fromPicks(numTeams, matches, results)
      } else {
        tree = MatchTree.fromMatchRes(numTeams, matches)
      }
      if (tree) {
        setMatchTree?.(tree)
      } else {
        console.error('no tree')
      }
    } else {
      console.error('no bracket')
    }
  }, [])

  const complete = matchTree && matchTree.allPicked()

  const handleUpdatePicks = () => {
    if (matchTree) {
      const picks = matchTree.toMatchPicks()
      if (!picks || picks.length === 0) return
      const complete = matchTree.allPicked()
      const data = {
        results: picks,
        shouldNotifyResultsUpdated: notifyParticipants,
      }
      bracketApi
        .updateBracket(bracketId, data)
        .then(() => {
          const dashboardUrl = getDashboardPath('hosting', 'closed')
          if (dashboardUrl) window.location.href = dashboardUrl
        })
        .catch((err) => {
          console.error(err)
          logger.error(err)
        })
    }
  }

  if (showPaginated) {
    return (
      <BracketResultsBuilderContext.Provider
        value={{ notifyParticipants, toggleNotifyParticipants }}
      >
        <PaginatedResultsBuilder
          {...props}
          handleUpdatePicks={handleUpdatePicks}
        />
      </BracketResultsBuilderContext.Provider>
    )
  }
  return (
    <div
      className={`wpbb-reset tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover ${
        darkMode ? ' tw-dark' : ''
      }`}
      style={{
        backgroundImage: `url(${darkMode ? darkBracketBg : lightBracketBg})`,
      }}
    >
      {matchTree && (
        <div
          className={`tw-flex tw-flex-col tw-items-center tw-max-w-screen-lg tw-m-auto tw-pb-[57px]`}
        >
          <div className="tw-py-[116px]">
            <ResultsBracket matchTree={matchTree} setMatchTree={setMatchTree} />
          </div>
          <div
            className={`tw-flex tw-flex-col tw-gap-24 ${
              !complete ? ' tw-max-w-[470px] tw-w-full' : ''
            }`}
          >
            <ActionButton
              variant="green"
              size="big"
              onClick={handleUpdatePicks}
            >
              {complete ? 'Complete Bracket' : 'Update Picks'}
            </ActionButton>
            <div className="tw-flex tw-gap-20 tw-items-center tw-self-center">
              <Checkbox
                id="notify-participants-check"
                checked={notifyParticipants}
                onChange={() => setNotifyParticipants(!notifyParticipants)}
              />
              <label
                htmlFor="notify-participants-check"
                className="tw-font-500 tw-text-24"
              >
                Notify Participants
              </label>
            </div>
          </div>
        </div>
      )}
    </div>
  )
}

const Wrapped = WithDarkMode(
  WithMatchTree(WithBracketMeta(BracketResultsBuilder))
)

export default Wrapped
