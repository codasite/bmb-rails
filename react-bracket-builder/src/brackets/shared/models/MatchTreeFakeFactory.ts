import { MatchTree } from './MatchTree'

export function fourTeamMatchTree() {
  const matches = [
    [
      {
        id: 9,
        roundIndex: 0,
        matchIndex: 0,
        team1: { id: 17, name: 'Team 1' },
        team2: { id: 18, name: 'Team 2' },
        team1Wins: true,
      },
      {
        id: 10,
        roundIndex: 0,
        matchIndex: 1,
        team1: { id: 19, name: 'Team 3' },
        team2: { id: 20, name: 'Team 4' },
        team1Wins: true,
      },
    ],
    [{ roundIndex: 1, matchIndex: 1, team2Wins: true }],
  ]
  return MatchTree.deserialize({ rounds: matches })
}
