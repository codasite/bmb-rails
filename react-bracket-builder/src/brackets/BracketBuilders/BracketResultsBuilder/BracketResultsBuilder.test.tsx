// DO NOT REMOVE REACT IMPORT. Needed for image generator
import React from 'react'
import BracketResultsBuilder from './BracketResultsBuilder'
import { fourTeamMatchTree } from '../../shared/models/MatchTreeFakeFactory'
import { render } from '@testing-library/react'

describe('BracketResultsBuilder', () => {
  test('renders page correctly', () => {
    const noop = () => {}
    const { asFragment } = render(
      <BracketResultsBuilder
        matchTree={fourTeamMatchTree()}
        setMatchTree={noop}
        bracket={{
          numTeams: 4,
          matches: [],
          results: [],
          id: 0,
        }}
        setBracketMeta={noop}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
