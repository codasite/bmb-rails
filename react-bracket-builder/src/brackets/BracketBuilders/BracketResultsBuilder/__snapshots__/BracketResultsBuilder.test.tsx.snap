// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BracketResultsBuilder renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="wpbb-reset tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover  tw-dark"
    style="background-image: url(undefined);"
  >
    <div
      class="tw-flex tw-flex-col tw-items-center tw-max-w-screen-lg tw-m-auto tw-pb-[57px]"
    >
      <div
        class="tw-py-[116px]"
      >
        <div
          class="tw-flex tw-flex-col tw-dark wpbb-default-bracket tw-relative"
        >
          <div
            class="tw-flex tw-flex-col tw-justify-end"
            style="margin-bottom: 95px; min-height: 0;"
          >
            <div
              class="tw-flex tw-flex-col tw-items-center"
              style="gap: 24px;"
            >
              <div
                class="wpbb-team-1-0-winner tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[297px] tw-h-[52px] tw-text-dd-blue tw-font-700 tw-bg-green"
                data-testid="team-slot-round-1-match-0-winner"
                style="border-width: 0px;"
              >
                <span
                  style="font-size: 32px; transform: scale(1); visibility: hidden;"
                >
                  Team 3
                </span>
              </div>
            </div>
          </div>
          <div
            class="tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-100"
          >
            <div
              class="tw-flex tw-justify-between"
              style="width: 700px;"
            >
              <div
                class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
              >
                <div
                  class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
                >
                  <button
                    class="wpbb-team-0-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                    data-testid="team-slot-round-0-match-0-left"
                    style="border-width: 0px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 1
                    </span>
                  </button>
                  <button
                    class="wpbb-team-0-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                    data-testid="team-slot-round-0-match-0-right"
                    style="border-width: 2px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 2
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[236px]"
              >
                <div
                  class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[-48px]"
                >
                  <button
                    class="wpbb-team-1-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                    data-testid="team-slot-round-1-match-0-left"
                    style="border-width: 2px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 1
                    </span>
                  </button>
                  <button
                    class="wpbb-team-1-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                    data-testid="team-slot-round-1-match-0-right"
                    style="border-width: 0px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 3
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
              >
                <div
                  class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
                >
                  <button
                    class="wpbb-team-0-1-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                    data-testid="team-slot-round-0-match-1-left"
                    style="border-width: 0px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 3
                    </span>
                  </button>
                  <button
                    class="wpbb-team-0-1-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                    data-testid="team-slot-round-0-match-1-right"
                    style="border-width: 2px;"
                  >
                    <span
                      style="font-size: 16px; transform: scale(1); visibility: hidden;"
                    >
                      Team 4
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            style="margin-top: 20px; min-height: 0;"
          >
            <div
              class="tw-flex tw-flex-col tw-gap-20 tw-justify-between tw-items-center tw-font-700 tw-whitespace-nowrap "
            >
              <span
                class="tw-text-36 tw-text-dd-blue dark:tw-text-white"
              >
                Who You Got?
              </span>
              <svg
                class="tw-w-[124px] tw-text-black/25 dark:tw-text-white"
              />
            </div>
          </div>
          <div
            class="tw-absolute"
          />
          <div
            class="tw-absolute"
          />
        </div>
      </div>
      <div
        class="tw-flex tw-flex-col tw-gap-24 "
      >
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-cursor-pointer disabled:tw-cursor-default tw-leading-none tw-cursor-pointer tw-bg-green/15 tw-text-white tw-gap-10 tw-text-36 tw-font-700 tw-rounded-8 tw-px-30 tw-py-16 tw-border-solid tw-border-4 tw-border-green"
        >
          Complete Bracket
        </button>
        <div
          class="tw-flex tw-gap-20 tw-items-center tw-self-center"
        >
          <input
            checked=""
            class="tw-appearance-none tw-rounded-8 tw-cursor-pointer tw-bg-white tw-bg-no-repeat tw-bg-center"
            id="notify-participants-check"
            style="background-image: url(); height: 32px; width: 32px;"
            type="checkbox"
          />
          <label
            class="tw-font-500 tw-text-24"
            for="notify-participants-check"
          >
            Notify Participants
          </label>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
